doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+
+sonic# show event start 2021-05-19T21:39:31.622Z end 2021-05-19T21:46:14.371Z
+----------------------------------------------------------------------------------------------------------------------------
+Id           Action          Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+3            CLEAR           CRITICAL   DUMMY_ALARM                    2021-05-19T21:42:34.371Z    :- signalHandler: Clearing simulated alarm 
+
+sonic# show alarm [ acknowledged | all | detail | summary | severity <sev> | id <seq-id> | start <from-ts> end <to-ts> | recent <5min|1hr|1day> | from <from-seq> to <to-seq> ]
+
+'show alarm' command would display all the *active* alarm records in ALARM table. Acknowledged alarms wont be shown here.
+
+sonic# show alarm
+----------------------------------------------------------------------------------------------------------------------------
+Id           Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+14           WARNING    TEMPERATURE_EXCEEDED           2021-05-20T00:47:52.992Z    :- temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees         
+16           WARNING    PSU_FAULT                      2021-05-20T02:16:42.611Z    :- /psu/2 has experienced a fault 
+
+sonic# show alarm all
+----------------------------------------------------------------------------------------------------------------------------
+Id           Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+14           WARNING    TEMPERATURE_EXCEEDED           2021-05-20T00:47:52.992Z    :- temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees         
+15           WARNING    DUMMY_ALARM                    2021-05-20T02:16:41.637Z    :- signalHandler: Raising simulated alarm         
+16           WARNING    PSU_FAULT                      2021-05-20T02:16:42.611Z    :- /psu/2 has experienced a fault 
+  
+
+sonic# show alarm detail 
+  
+alarm details - 14
+-------------------------------------------
+Id:                14
+Severity:          CRITICAL
+Source:            /sensor/2
+Name:              TEMPERATURE_EXCEEDED
+Description:       temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+Raise-time:        Wed Feb 10 18:08:24 2021
+Ack-time:          
+New:               true
+Acknowledged:      false
+
+sonic# show alarm from 14 to 16
```

Comment:
Same question for range in event

Reply:
There is no range as the sequence-id can grow until 64 bit.

Reply:
what i am referring is check for validity such as if the user provides 16 to 14, will you check and provide proper error notification? is 14 to 14 is allowed? Just add a comment what type of validation and behaviour is expected.

Reply:
@liat-grozovik - HLD should not have to cover each & every validation that will be added. Isn't it ?